src/Rector/v9/v4/UseContextApiForVersioningWorkspaceIdRector.php

```diff
@@ -49,7 +49,7 @@ public function refactor(Node $node): ?Node
             return null;
         }
 
-        $parentNode = $node->getAttribute('parent');
+        $parentNode = $node->getAttribute(AttributeKey::PARENT_NODE);
```

Comment:
Strings attribute can typoed, eg. `parnet` (been there :)) and are hard to detect.
With constants, there is a future proof that even if the string names changes, the constant will still work.

Reply:
@TomasVotruba Most of the time i did it right. But sometimes the evenings were so long. You know ;-)

Reply:
I know, same here :)

That's why I've delegated this task to PHPStan:

```neon
services:
    # require constant in argument position
    -
        class: Symplify\PHPStanRules\Rules\RequireMethodCallArgumentConstantRule
        tags: [phpstan.rules.rule]
        arguments:
            constantArgByMethodByType:
                PhpParser\Node:
                    getAttribute: [0]
                    hasAttribute: [0]
                    setAttribute: [0]
```

He will tell me if something is wrong :+1: